### PR TITLE
Add documentation about the Minikube extension

### PR DESCRIPTION
**Pull request by geoand**
Follows up on #8840

### CODE REVIEW

File: docs/src/main/asciidoc/deploying-to-kubernetes.adoc [hunk 1]
```diff
@@ -388,12 +388,53 @@ To access the API server from within a Kubernetes cluster, some RBAC related res
...
+
+https://github.com/kubernetes/minikube[Minikube] is quite popular when a Kubernetes cluster is needed for development purposes. To make the deployment to Minikube
+experience as frictionless as possible, Quarkus provides the `quarkus-minikube` extension. This extension can be added to a project like so:
+
+[source,xml]
+----
+    <dependency>
+      <groupId>io.quarkus</groupId>
+      <artifactId>quarkus-minikube</artifactId>
+    </dependency>
```
Comment (gsmet) [inline docs/src/main/asciidoc/deploying-to-kubernetes.adoc:401]: It's minor but, to be consistent with the rest of the doc, let's remove the leading indent and have `groupId` and `artifactId` indented with 4 spaces.

Comment (gsmet) [inline docs/src/main/asciidoc/deploying-to-kubernetes.adoc:401]:
```suggestion
<dependency>
    <groupId>io.quarkus</groupId>
    <artifactId>quarkus-minikube</artifactId>
</dependency>
```

It's minor but, to be consistent with the rest of the doc, let's remove the leading indent and have `groupId` and `artifactId` indented with 4 spaces.

File: docs/src/main/asciidoc/deploying-to-kubernetes.adoc [hunk 2]
```diff
@@ -388,12 +388,53 @@ To access the API server from within a Kubernetes cluster, some RBAC related res
...
+----
+
+The purpose of this extension is to generate Kubernetes manifests (`minikube.yaml` and `minikube.json`) that are tailored to Minikube.
+This extension assumes a couple things:
+
+* Users won't be using an image registry and will instead make their container image accessible to the Kubernetes cluster by building it directly
+into Minikube's Docker daemon. To use Minikube's Docker daemon you must first execute:
+
```
Comment (gsmet) [inline docs/src/main/asciidoc/deploying-to-kubernetes.adoc:409]: I think you need a `+` here to make the source part of the bullet point.

File: docs/src/main/asciidoc/deploying-to-kubernetes.adoc [hunk 3]
```diff
@@ -388,12 +388,53 @@ To access the API server from within a Kubernetes cluster, some RBAC related res
...
+[source]
+----
+eval $(minikube -p minikube docker-env)
+----
+
+* Applications deployed to Kubernetes won't be accessed via a Kubernetes `Ingress`, but rather as a `NodePort` `Service`.
+The advantage of doing this is that the URL of an application can be retrieved trivially by executing:
+
```
Comment (gsmet) [inline docs/src/main/asciidoc/deploying-to-kubernetes.adoc:417]: Same here.

File: docs/src/main/asciidoc/deploying-to-kubernetes.adoc [hunk 4]
```diff
@@ -388,12 +388,53 @@ To access the API server from within a Kubernetes cluster, some RBAC related res
...
+[source]
+----
+minikube service list
+----
+
+To control the https://kubernetes.io/docs/concepts/services-networking/service/#nodeport[nodePort] that is used in this case, users can set `quarkus.kubernetes.node-port`.
+Note however that this configuration is entirely optional because Quarkus will automatically use a proper (and non-changing) value if none is set.
+
+WARNING: It is highly discouraged to use the manifests generated by the Minikube extension when deploying to production as these manifests are intended for development purposes
+only. When deploying to production, consider using the vanilla Kubernetes manifests (or the Openshift ones when targeting Openshift).
```
Comment (gsmet) [inline docs/src/main/asciidoc/deploying-to-kubernetes.adoc:427]:
```suggestion
only. When deploying to production, consider using the vanilla Kubernetes manifests (or the OpenShift ones when targeting OpenShift).
```

File: docs/src/main/asciidoc/deploying-to-kubernetes.adoc [hunk 5]
```diff
@@ -734,6 +775,19 @@ The generated service can be customized using the following properties:
...
+=== Deployment targets
+
+Mentioned in the previous sections was the concept of `deployment-target`. This concept allows users to control which Kubernetes manifests will be generated
+and deployed to a cluster (if `quarkus.kubernetes.deploy` has been set to `true`).
+
+By default, when no `deployment-target` is set, then only vanilla Kubernetes resources are generated and deployed. When multiple values are set (for example
+`quarkus.kubernetes.deployment-target=kubernetes,openshift`) then the resources for all targets are generated, but only the resources
+that correspond to the *first* target are applied to the cluster (if deployment is enabled).
+
+In the case of wrapper extension like Openshift and Minikube, when these extensions have been explicitly added to the project, the default `deployment-target`
```
Comment (gsmet) [inline docs/src/main/asciidoc/deploying-to-kubernetes.adoc:787]:
```suggestion
In the case of wrapper extensions like OpenShift and Minikube, when these extensions have been explicitly added to the project, the default `deployment-target`
```

**Review by gsmet — CHANGES_REQUESTED** [review]
Added a couple suggestions.

File: docs/src/main/asciidoc/deploying-to-kubernetes.adoc (hunk 2 above)
Comment (geoand) [reply docs/src/main/asciidoc/deploying-to-kubernetes.adoc]: I didn't know that trick, thanks!

**Review by gsmet — APPROVED** [review]

**Pull request merged**